File: Core Java/shoppingcart-app/src/com/techlabs/shoppingcart/Order.java
package com.techlabs.shoppingcart;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class Order {

	private int id;
	private Date orderDate;
	private List<LineItem> items = new ArrayList<LineItem>();;
	private double checkoutCost=0;

	public Order(int id,Date orderDate) {

		this.id=id;
		this.orderDate=orderDate;
	}
	
	public void addItem(LineItem item)
	{
		items.add(item);
	}
	
	public double calculateCheckout()
	{
		for(LineItem item:items)
		{
			checkoutCost = checkoutCost + item.costOfLineItem();
		}
		return checkoutCost;
	}

	public List<LineItem> getItems() {
		return items;
	}

	public int getId() {
		return id;
	}

	public Date getOrderDate() {
		return orderDate;
	}
	
	public double getCheckoutCost() {
		return checkoutCost;
	}
}

File: DOTNET/C#/ShoppingCartApp/ShoppingCartApp/Order.cs
using System;
using System.Collections.Generic;

namespace ShoppingCartApp
{
    public class Order
    {
        private readonly int id;
        private readonly DateTime orderDate;
        private List<LineItem> items = new List<LineItem>();
        private double checkoutCost = 0;

        public Order(int id, DateTime orderDate)
        {
            this.id = id;
            this.orderDate = orderDate;
        }

        public void AddItem(LineItem item)
        {
            items.Add(item);
        }

        public double CalculateCheckout()
        {
            foreach (LineItem item in items)
            {
                checkoutCost = checkoutCost + item.CostOfLineItem();
            }
            return checkoutCost;
        }

        public List<LineItem> Items { get { return items; } }
        public int Id { get { return id; } }
        public DateTime OrderDate { get { return orderDate; } }
        public double CheckoutCost { get { return checkoutCost; } }
    }
}
